### PR TITLE
feat: fetch data hook for new sHakka balance

### PR DESCRIPTION
**Pull request by raman-pelith**
add new sHakka abi
feat: feat: fetch data hook for new sHakka balance

### CODE REVIEW

File: src/hooks/useSHakkaBalance.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    const [transactionSuccess, setTransactionSuccess] = useState(false);
+  
+    const fetchDataState: ChainDataFetchingState = useMemo(() => {
+      return transactionSuccess ? ChainDataFetchingState.SUCCESS : ChainDataFetchingState.LOADING;
+    }, [transactionSuccess]);
+  
+    const ethProvider = new JsonRpcProvider(process.env.REACT_APP_NETWORK_URL);
+    const bscProvider = new JsonRpcProvider(process.env.REACT_APP_BSC_NETWORK_URL);
+    const polygonProvider = new JsonRpcProvider(process.env.REACT_APP_POLYGON_NETWORK_URL);
+    const kovanProvider = new JsonRpcProvider(process.env.REACT_APP_KOVAN_NETWORK_URL);
```
Comment (paul-pelith) [inline src/hooks/useSHakkaBalance.ts:32]: after deep in source code, there has promise asynchronous process on contructor, maybe should add these to useRef / useMemo, prevent unhandled error?

Comment (ytwang00) [reply src/hooks/useSHakkaBalance.ts:32]: I think it is a good idea, cause this code will new the provider every rerender

File: src/hooks/useSHakkaBalance.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    const fetchSHakkaBalance = async () => {
+      setTransactionSuccess(false);
+      const ethMulticallProvider = new MulticallProvider(ethProvider, ChainId.MAINNET);
```
Comment (ytwang00) [inline src/hooks/useSHakkaBalance.ts:36]: provider => contract => balance could make a function, don't repeat so many times

File: src/hooks/useSHakkaBalance.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      const bscMulticallProvider = new MulticallProvider(bscProvider, ChainId.BSC);
+      const polygonMulticallProvider = new MulticallProvider(polygonProvider, ChainId.POLYGON);
+      const kovanMulticallProvider = new MulticallProvider(kovanProvider, ChainId.KOVAN);
+
+      const ethSHakkaContract = new MulticallContract(NEW_SHAKKA_ADDRESSES[ChainId.MAINNET], STAKING_ABI);
+      const bscSHakkaContract = new MulticallContract(NEW_SHAKKA_ADDRESSES[ChainId.BSC], STAKING_ABI);
+      const polygonSHakkaContract = new MulticallContract(NEW_SHAKKA_ADDRESSES[ChainId.POLYGON], STAKING_ABI);
+      const kovanSHakkaContract = new MulticallContract(NEW_SHAKKA_ADDRESSES[ChainId.KOVAN], STAKING_ABI);
```
Comment (ytwang00) [inline src/hooks/useSHakkaBalance.ts:44]: They are all single call, why need use multical?

File: src/hooks/useSHakkaBalance.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        console.log(e);
+        console.log('fetch user sHakka balance error');
+      }
+    };
+  
+    const debouncedFetchSHakkaBalance = useMemo(() => debounce(fetchSHakkaBalance, 200), [fetchSHakkaBalance]);
+  
+    useEffect(() => {
+      debouncedFetchSHakkaBalance();
+    }, [ethProvider, bscProvider, polygonProvider, kovanProvider, latestBlockNumber]);
```
Comment (ytwang00) [inline src/hooks/useSHakkaBalance.ts:68]: ethProvider, bscProvider, polygonProvider, kovanProvider will change every rerender, i think it means this useEffect will be triggered too many times

**Review by ytwang00 — CHANGES_REQUESTED** [review]

File: src/hooks/useSHakkaBalance.ts [hunk 5]
```diff
@@ -25,29 +25,29 @@ export default function useSHakkaBalance(): {
...
+    };
 
+    const fetchSHakkaBalance = async () => {
+      setTransactionSuccess(false);
       try {
-        const [ ethSHakkaBalance ] = await ethMulticallProvider.all([ethSHakkaContract.balanceOf(account)]);
-        const [ bscSHakkaBalance ] = await bscMulticallProvider.all([bscSHakkaContract.balanceOf(account)]);
-        const [ polygonSHakkaBalance ] = await polygonMulticallProvider.all([polygonSHakkaContract.balanceOf(account)]);
-        const [ kovanSHakkaBalance ] = await kovanMulticallProvider.all([kovanSHakkaContract.balanceOf(account)]);
+        const ethSHakkaBalance = await getSHakkaBalance(ChainId.MAINNET);
```
Comment (ytwang00) [inline src/hooks/useSHakkaBalance.ts:47]: await promise.all

**Pull request merged**